### PR TITLE
Toc cypress

### PR DESCRIPTION
**Pull request by steffenri**

### CODE REVIEW

File: cypress/integration/blocks.js [hunk 1]
```diff
@@ -596,5 +596,35 @@ if (Cypress.env('API') !== 'guillotina') {
     //   //   cy.contains(expected);
     //   // }
     // });
+
+    it.only('Add Table of Contents block', () => {
+      const h2 = 'expectedH2';
```
Comment (tisto) [inline cypress/integration/blocks.js]: @steffenri I prefer to be explicit here and not use a variable.

File: cypress/integration/blocks.js [hunk 2]
```diff
@@ -596,5 +596,35 @@ if (Cypress.env('API') !== 'guillotina') {
...
+
+      //Add H2
+      cy.get('.block.inner.text .public-DraftEditor-content')
+        .type(` ${h2}`)
+        .setSelection(` ${h2}`);
```
Comment (tisto) [inline cypress/integration/blocks.js]: @steffenri why is .setSelection necessary here?

File: cypress/integration/blocks.js [hunk 3]
```diff
@@ -596,5 +596,35 @@ if (Cypress.env('API') !== 'guillotina') {
...
+        .click();
+      cy.get('.ui.basic.icon.button.toc')
+        .contains('Table of Contents')
+        .click();
+
+      //Save
+      cy.get('#toolbar-save').click();
+
+      //View
+      cy.get(' a').contains(` ${h2}`);
```
Comment (tisto) [inline cypress/integration/blocks.js]: @steffenri this checks any "a" element. We should check the output of the TOC block for the h2 headline.

File: cypress/integration/blocks.js (hunk 2 above)
Comment (steffenri) [reply cypress/integration/blocks.js]: using .setSelection here allows us to style in draftJS. In this case we can make the selected text a h2.

File: cypress/integration/blocks.js (hunk 1 above)
Comment (steffenri) [reply cypress/integration/blocks.js]: changed.

File: cypress/integration/blocks.js (hunk 3 above)
Comment (steffenri) [reply cypress/integration/blocks.js]: Fixed that aswell. We now check for an a element in the list that gets created by the TOC Block. We also check for the exact header type.

**Pull request merged**